lib/rpc.js

```diff
@@ -45,31 +48,41 @@ module.exports = class RPC {
     return this.socket.address()
   }
 
-  bind (port) {
-    return new Promise((resolve, reject) => {
-      const s = this.socket
+  bind (port = this._bind) {
+    if (this._binding) return this._binding
 
-      if (s.listen) {
-        s.listen(port)
-      } else {
-        s.bind(port)
-      }
+    const defaultPort = this._bind
+
+    this._binding = new Promise((resolve, reject) => {
+      const s = this.socket
 
+      s.bind(port)
       s.on('listening', onlistening)
       s.on('error', onerror)
 
       function onlistening () {
+        this._bound = true
+
         s.removeListener('listening', onlistening)
         s.removeListener('error', onerror)
         resolve()
```

Comment:
if we resolve the promise with the port, then `bind()` will return a promise that resolves to port as a sort confirmation, we could also use this internally to eventuall set _binding to the actual port (and then for debugging you can check that _bind and _binding match (unless _bind is 0)  (eg `return (this,_binding = await this._binding)`)

Reply:
great idea, will make it resolve to the port